src/watchdog/utils/dirsnapshot.py

```diff
@@ -220,7 +220,7 @@ def walk(root):
                 # list of its parent and trying to delete its contents. If this
                 # happens we treat it as empty. Likewise if the directory was replaced
                 # with a file of the same name (less likely, but possible).
-                if e.errno == errno.ENOENT or e.errno == errno.ENOTDIR:
+                if e.errno in (errno.ENOENT, errno.ENOTDIR, errno.EINVAL):
```

Comment:
Looks like this was be merged in #559?

Reply:
Yes, my branch is not up-to-date with master I guess. Will rebase it.